pdr_backend/util/web3_accounts.py

```diff
@@ -0,0 +1,98 @@
+import sys
+from os import getenv
+from typing import List
+from enforce_typing import enforce_types
+
+from eth_account import Account
+from pdr_backend.ppss.web3_pp import Web3PP
+from pdr_backend.contract.token import NativeToken, Token
+from pdr_backend.util.contract import get_address
+from pdr_backend.util.mathutil import to_wei, from_wei
+
+
+@enforce_types
+def create_accounts(n_accounts: int):
+    """
+    @description
+        Create new accounts using the web3 provider and print the private_key and the public_key
+    """
+    # loop through n_wallets
+    for i in range(n_accounts):
+        # pylint: disable=no-value-for-parameter
+        new_account = Account.create()
+        print(f"\nWallet #{i}")
+        print(f"Private Key: {new_account._private_key.hex()}")
+        print(f"Public Key: {new_account.address}")
+
+
+@enforce_types
+def _get_account_balances(checksum_address: str, web3_pp: Web3PP):
+    """
+    @description
+        Get account balances from an address
+    """
+    # loop through n_wallets
+    native_token = NativeToken(web3_pp)
+    OCEAN_addr = get_address(web3_pp, "Ocean")
+    OCEAN_token = Token(web3_pp, OCEAN_addr)
+
+    native_token_balance = native_token.balanceOf(checksum_address)
+    OCEAN_balance = OCEAN_token.balanceOf(checksum_address)
+
+    return native_token_balance, OCEAN_balance
+
+
+@enforce_types
+def view_accounts(addresses: List[str], web3_pp: Web3PP):
+    """
+    @description
+        View account balances for multiple addresses
+    """
+    # loop through n_wallets
+    for address in addresses:
+        checksum_address = web3_pp.web3_config.w3.to_checksum_address(address.lower())
+        native_token_balance, OCEAN_balance = _get_account_balances(
+            checksum_address, web3_pp
+        )
+
+        print(f"Account {checksum_address}")
+        print(f"Native token balance: {from_wei(native_token_balance)}")
+        print(f"OCEAN balance: {from_wei(OCEAN_balance)}")
+
+
+@enforce_types
+def fund_accounts(
+    amount: float, to_addressess: List[str], web3_pp: Web3PP, is_native_token: bool
+):
+    """
+    @description
+        Fund multiple accounts using native or OCEAN tokens.
+        Sends a total of (Amount * n_accounts).
+    """
+    if web3_pp.network not in ["sapphire-testnet", "sapphire-mainnet"]:
+        print("Unknown network")
+        sys.exit(1)
+
+    token = None
+    token_name = None
+    if is_native_token:
```

Comment:
I suggest adding a name property as a helper in NativeToken and OceanToken classes. I think it can be useful in the future.

Then:
- the if only decides "a single thing"
- later in print f"Sending {token.name} token to {address} for the amount of {amount} wei"

Reply:
Updated